pkg/webhook/controlplane/ensurer.go

```diff
@@ -357,92 +284,42 @@ var (
 	}
 )
 
-func ensureKubeAPIServerVolumeMounts(c *corev1.Container, csiEnabled, csiMigrationComplete bool) {
-	if csiEnabled && csiMigrationComplete {
-		c.VolumeMounts = extensionswebhook.EnsureNoVolumeMountWithName(c.VolumeMounts, cloudProviderConfigVolumeMount.Name)
-		c.VolumeMounts = extensionswebhook.EnsureNoVolumeMountWithName(c.VolumeMounts, cloudProviderSecretVolumeMount.Name)
-		return
-	}
-
-	c.VolumeMounts = extensionswebhook.EnsureVolumeMountWithName(c.VolumeMounts, cloudProviderConfigVolumeMount)
-	c.VolumeMounts = extensionswebhook.EnsureVolumeMountWithName(c.VolumeMounts, cloudProviderSecretVolumeMount)
+func ensureKubeAPIServerVolumeMounts(c *corev1.Container) {
+	c.VolumeMounts = extensionswebhook.EnsureNoVolumeMountWithName(c.VolumeMounts, cloudProviderConfigVolumeMount.Name)
+	c.VolumeMounts = extensionswebhook.EnsureNoVolumeMountWithName(c.VolumeMounts, cloudProviderSecretVolumeMount.Name)
```

Comment:
Assuming that all clusters are now running at least K8s 1.20 and the CSI migration was complete for them in 1.18 we can clean up these `EnsureNoVolumeMountWithName` calls.

Previously the handling was:
https://github.com/gardener/gardener-extension-provider-gcp/blob/38db4bac5eadeba6b5aa7f4763e09fe87cde4416/pkg/webhook/controlplane/ensurer.go#L360-L369

Prior to CSI we were ensuring volume and volume mount for the cloud-provider-config ConfigMap for components like the kube-apiserver, kube-controller-manager. With CSI we no longer need this volume and volume mount, hence we were ensuring that any potential present volume/volume mount gets cleaned up after the update to a version that enables CSI.
I think it should be safe to drop these `EnsureNoVolumeMountWithName` calls for the cloud-provider-config ConfigMap as they should be already cleaned up from the kube-apiserver/kube-controller-manager deployments after CSI migration is complete. And today there shouldn't be any handling that adds these cloud-provider-config volume and volume mount.